### PR TITLE
[test-integration] Add a registry package with registry v1/v2 code

### PR DESCRIPTION
**Pull request by vdemeester**
This extract what was in registry_test.go and registry_mock_test.go. This also move `RegistryHosting` requirement to `registry.Hosting` 🦁.

Hopefully, my comments are ok 👼

/cc @thaJeztah @icecrime @dnephin @cpuguy83 @tiborvass @AkihiroSuda @aaronlehmann @dmcgowan 

🐸

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

File: integration-cli/docker_cli_pull_local_test.go [hunk 1]
```diff
@@ -347,7 +347,8 @@ func (s *DockerRegistrySuite) TestPullManifestList(c *check.C) {
 	manifestListDigest := digest.FromBytes(manifestListJSON)
 	hexDigest := manifestListDigest.Hex()
 
-	registryV2Path := filepath.Join(s.reg.dir, "docker", "registry", "v2")
+	//registryV2Path := filepath.Join(s.reg.dir, "docker", "registry", "v2")
```
Comment (thaJeztah) [inline integration-cli/docker_cli_pull_local_test.go]: leftover comment?

**Review by thaJeztah — CHANGES_REQUESTED** [review]
found one leftover comment, but LGTM otherwise

File: integration-cli/docker_cli_pull_local_test.go (hunk 1 above)
Comment (vdemeester) [inline integration-cli/docker_cli_pull_local_test.go]: Yes 😅

File: integration-cli/registry/requirement.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package registry
+
+import "os/exec"
+
+// Hosting returns wether the host can host a registry (v2) or not
```
Comment (AkihiroSuda) [inline integration-cli/registry/requirement.go]: nit: wether -> whether

File: integration-cli/registry/registry_mock.go [hunk 3]
```diff
@@ -1,32 +1,33 @@
...
 	"strings"
 	"sync"
-
-	"github.com/go-check/check"
 )
 
 type handlerFunc func(w http.ResponseWriter, r *http.Request)
 
-type testRegistry struct {
+// Mock represent a registry mock
```
Comment (AkihiroSuda) [inline integration-cli/registry/registry_mock.go]: nit: represent -> represents

File: integration-cli/registry/registry_mock.go [hunk 4]
```diff
@@ -1,32 +1,33 @@
...
+type Mock struct {
 	server   *httptest.Server
 	hostport string
 	handlers map[string]handlerFunc
 	mu       sync.Mutex
 }
 
-func (tr *testRegistry) registerHandler(path string, h handlerFunc) {
+// RegisterHandler register the specified handler for the registry mock
```
Comment (AkihiroSuda) [inline integration-cli/registry/registry_mock.go]: nit: register -> registers

**Review by AkihiroSuda — APPROVED** [review]
LGTM

(typos are trivial)

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**